### ISSUE TITLE
[Alert Dialog] Padding added on bottom when the positive button's text is a little long

### ISSUE THREAD

**Issue by aryanarora180** [material-components/material-components-android#1669]
**Description:** When creating a MaterialAlertDialog with the positive button's text a little long, padding gets added to the bottom of the dialog making it look out of place.

![Screenshot_20200831-130148](https://user-images.githubusercontent.com/28786413/91697512-e189a880-eb8e-11ea-8643-98acb56180cf.jpg)

**Expected behavior:** There should be no padding added.

![Screenshot_20200831-133918](https://user-images.githubusercontent.com/28786413/91698206-f7e43400-eb8f-11ea-9fa7-c8b29e9bfb09.jpg)

**Source code:**
```Kotlin
MaterialAlertDialogBuilder(requireContext()).apply {
                    setTitle("Dialog title")
                    setMessage("Dialog body")
                    setPositiveButton("Agree and continue") { _, _ ->
                        
                    }
                    setNeutralButton("No") { _, _ ->
                        
                    }
                    show()                    
                }
```

**Devices:** OnePlus 6, Xiaomi Mi A2, Motorola G5 Plus
**Android API versions:** OnePlus 6 - 29, Xiaomi Mi A2 - 28, Motorola G5 Plus - 27
**Material Library version:** 1.2.0

**Comment by leticiarossi** [comment]
Hi, I'm not able to reproduce it, could you provide a sample app where the issue happens?

**Comment by aryanarora180** [comment]
I have created a [repository](https://github.com/aryanarora180/AlertDialog). I made it completely from scratch, only added Material components dependency in the gradle files and changed the app theme to inherit from MaterialComponents. Then I displayed the dialog.

**Comment by gabrielemariotti** [comment]
The issue can be replicated using an emulator with these specs:
https://www.gsmarena.com/oneplus_6-9109.php

<img width="299" alt="Schermata 2020-09-01 alle 09 50 57" src="https://user-images.githubusercontent.com/2583078/91822847-a479f080-ec38-11ea-9b9e-d465979e0af1.png">

<img width="971" alt="Schermata 2020-09-01 alle 09 51 49" src="https://user-images.githubusercontent.com/2583078/91822951-cd9a8100-ec38-11ea-866e-8428998d0820.png">

It doesn't happen with the appcompat `AlertDialog.Builder`.

**Comment by ivannarino** [comment]
It does also appear on AlertDialog.Builder, just make the buttons labels larger and you can reproduce

![image](https://user-images.githubusercontent.com/5160934/95664586-fe5eb780-0b0e-11eb-9508-debd4e19c24a.png)

```
AlertDialog.Builder(this)
                .setTitle("Discard draft?")
                .setMessage("This will permanently delete the current e-mail draft.")
                .setPositiveButton("Agree and continueeee") { _, _ -> }
                .setNeutralButton("No") { _, _ -> }
                .show()
```

**Comment by ivannarino** [comment]
And depending on the version, we have different behaviors:

### v19 extra top padding on the button panel
![image](https://user-images.githubusercontent.com/5160934/95665128-eee16d80-0b12-11eb-9f62-1077e9d680af.png)


### v29 parent panel has extra height
![image](https://user-images.githubusercontent.com/5160934/95665101-bcd00b80-0b12-11eb-8760-b3ba5ba44813.png)

When there is enough space for the buttons, the dialogs look like this:

### v19
![image](https://user-images.githubusercontent.com/5160934/95665239-18e75f80-0b14-11eb-9c77-00185fa59353.png)

### v29
![image](https://user-images.githubusercontent.com/5160934/95665240-20a70400-0b14-11eb-93f6-d0570d38b63e.png)

**Comment by aryanarora180** [comment]
Good find! While I wasn't able to reproduce the same thing for the AlertDialog.Builder, this shows that there is inconsistency across all types of devices.

**Comment by pekingme** [comment]
Thanks for all the details to reproduce this. This bug could be resolved when we migrated to M3 MaterialAlertDialog. I have observed any described behaviors in M3 styles. 

I'm closing this issue. Feel free to reopen if this persists in M3 themes/styles.

**Comment by kenshin171** [comment]
what about people who cannot migrate to M3 styles yet?

**Comment by drchen** [comment]
Hi, I believe this is caused by the measuring logic of AppCompat's ButtonBarLayout under a certain combination of button text, text appearance, and paddings. 

There's really not much we can do with the old M2 styles because we cannot change the style (and we should prevent doing that.)

As an workaround for M2 users, tweaking `?attr/textAppearanceButton` might be help - another option is overriding `?android:attr/layout` specified in `?attr/alertDialogStyle` and adjust the layout to workaround the issue.

**Comment by pekingme** [comment]
Verified bug existing in 1.6.0 and before, fixed in 1.7.0-alpha02. Also duplicate with #742

**Comment by pekingme** [comment]
Confirm this as a bug in appcompat lib. It's been fixed in appcompat 1.4 and later. MDC 1.7.0-alpha02 has upgraded appcompat to 1.4. If you are using earlier versions, add/update appcompat to 1.4 in your build.gradle should fix it.
